### PR TITLE
Stripe integration

### PR DESCRIPTION
**Pull request by iamthuypham**
# Description

Creating subscibe api using stripe

- [x] get subscription
- [x] subscribe to paid plan
- [x] cancel subscription
- [x] refactor state with redux
- [x] improve UI
- [x] fix test case

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: server/src/api/modules/auth.js [hunk 1]
```diff
@@ -11,11 +11,15 @@ export const register = (req, res) => {
...
+      let newUser = new User({
+        username,
+        password,
+        email
+      })
       newUser
         .save()
         .then(() => {
-          res.status(201).send("success")
+          res.status(201).send()
```
Comment (alexbotello) [inline server/src/api/modules/auth.js]: If you'd prefer not to send a confirmation message perhaps change to `res.sendStatus(201)`

File: server/src/api/modules/auth.js [hunk 2]
```diff
@@ -56,18 +60,18 @@ export const login = (req, res) => {
...
-    }
-    next()
-  })
+  // if (!req.headers.authorization) {
+  //   return res.status(400).send("Bad Request")
+  // }
+  // let token = req.headers.authorization
+  // token = token.replace("Bearer ", "")
+  // jwt.verify(token, JWT_SECRET, err => {
+  //   if (err) {
```
Comment (alexbotello) [inline server/src/api/modules/auth.js]: All this needs to be uncommented and added back please :)

File: server/src/api/resources/subscribe/subscribe.controller.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+export const subscribeToFreePlan = async (req, res, stripe) => {
+  const { source } = req.body
+  const customer = await stripe.customers.create({ source: source.id })
+
+  if (customer) {
+    const updatedRequest = {
+      ...req,
+      body: {
+        subscribed: true,
+        subdate: Date.now(),
```
Comment (alexbotello) [inline server/src/api/resources/subscribe/subscribe.controller.js]: Let's change all occurrences of `subdate` to `subDate` to align with changes made in my PR

File: server/tests/modules/testSubscribe.js [hunk 4]
```diff
@@ -3,66 +3,55 @@ import app from "../../src/server"
...
+let userId
+let token
+
+const userData = {
   username: "FreeUser",
   password: "freeUserPass",
   email: "email@hotmail.com",
   subscribed: false,
   customerId: "",
   subscribeId: ""
```
Comment (alexbotello) [inline server/tests/modules/testSubscribe.js]: These properties are still being set to `null` in the database right here.

**Comment by iamthuypham** [comment]
Alright @alexbotello. I cleared up the other failed tests. 

The last failed test, what I mentioned, is Stripe complained about api key, and also not show during my local testing.

So, it seems a little bit hard now to debugging, since the failure only appears in build.

Can you try updating the .env for the build? I sent to you directly in our group messenger.

**Pull request merged**